src/BaseFactory.php

```diff
@@ -51,8 +51,9 @@ public function __call(string $name, array $arguments): array
 
     /**
      * @param Attributes|string $attributes
+     * @final
      */
-    final public static function new(array|callable|string $attributes = [], string ...$states): static
+    public static function new(array|callable|string $attributes = [], string ...$states): static
     {
```

Comment:
this change is made to deprecate the usage of `PersistentObjectFactory` with "non-persistable" classes